docs/source/API/containers-index.rst

```diff
@@ -7,6 +7,8 @@ API: Containers
 
    * - Container
      - Description
+   * - `Array <containers/Array.html>`__
```

Comment:
I see this more like `View` ... do we really want this to be in containers?

Reply:
In C++ it is considered a container, but I don't have a strong feeling on it.

Reply:
Here Containers refers to the Kokkos subpackage, that is, everything that is defined in the `containers/src/` directory.  `Array` belongs to Core so it is not the right place.
The API reference for Core is a bit messy at the moment so I would either put it in Utilities next to `Complex` or in the STL Compat section.  We can put organize things better in follow ups.

Reply:
Moved to STL Compat.